src/Code/ExpressionOptimizer.fs

```diff
@@ -303,6 +319,95 @@ module Methods =
+            // x >= a && x <= a -> x = a
+            | ExpressionType.GreaterThanOrEqual, ExpressionType.LessThanOrEqual 
+            | ExpressionType.LessThanOrEqual, ExpressionType.GreaterThanOrEqual when propertyMatch leftVal rightVal ->
+                Expression.MakeBinary(ExpressionType.Equal, leftVar, leftVal) :> Expression
+            // Detect impossible ranges: x > a && x < b where a >= b
+            | ExpressionType.GreaterThan, ExpressionType.LessThan 
+            | ExpressionType.GreaterThanOrEqual, ExpressionType.LessThan
+            | ExpressionType.GreaterThan, ExpressionType.LessThanOrEqual ->
+                // For constant values, we can detect impossible ranges
+                match leftVal, rightVal with
+                | (:? ConstantExpression as leftConst), (:? ConstantExpression as rightConst) 
+                    when leftConst.Value :? IComparable && rightConst.Value :? IComparable ->
+                    let leftComp = leftConst.Value :?> IComparable
+                    let rightComp = rightConst.Value :?> IComparable
+                    if leftComp.CompareTo(rightComp) >= 0 then
+                        Expression.Constant(false, typeof<bool>) :> Expression
+                    else noHit
+                | _ -> noHit
+            | _ -> noHit
+        | Or' (ComparisonExpression (leftOp, leftVar, leftVal), ComparisonExpression (rightOp, rightVar, rightVal))
+            when propertyMatch leftVar rightVar ->
+            match leftOp, rightOp with  
+            // x > a || x > b -> x > min(a,b)
+            | ExpressionType.GreaterThan, ExpressionType.GreaterThan
+            | ExpressionType.GreaterThanOrEqual, ExpressionType.GreaterThan
+            | ExpressionType.GreaterThan, ExpressionType.GreaterThanOrEqual when propertyMatch leftVal rightVal ->
+                Expression.MakeBinary(ExpressionType.GreaterThan, leftVar, rightVal) :> Expression
+            | ExpressionType.GreaterThanOrEqual, ExpressionType.GreaterThanOrEqual when propertyMatch leftVal rightVal ->
+                Expression.MakeBinary(ExpressionType.GreaterThanOrEqual, leftVar, rightVal) :> Expression
+            // x < a || x < b -> x < max(a,b)
+            | ExpressionType.LessThan, ExpressionType.LessThan  
+            | ExpressionType.LessThanOrEqual, ExpressionType.LessThan
+            | ExpressionType.LessThan, ExpressionType.LessThanOrEqual when propertyMatch leftVal rightVal ->
+                Expression.MakeBinary(ExpressionType.LessThan, leftVar, rightVal) :> Expression
+            | ExpressionType.LessThanOrEqual, ExpressionType.LessThanOrEqual when propertyMatch leftVal rightVal ->
+                Expression.MakeBinary(ExpressionType.LessThanOrEqual, leftVar, rightVal) :> Expression
+            | _ -> noHit
+        | noHit -> noHit
+
+    /// Optimize string operations: str.Length > 0 -> !String.IsNullOrEmpty(str)
+    let ``optimize string operations`` = function
```

Comment:
This is so rare case that I think the checking costs more than the optimization benefits.